tasks/species_typing/task_tbprofiler_output_parsing.wdl

```diff
@@ -104,166 +151,386 @@ task tbprofiler_output_parsing {
-        return "The detected genetic determinant(s) have uncertain significance, resistance to {} cannot be ruled out".format(drug)
-      elif annotation == "Assoc w R - interim":
-        return "The detected genetic determinant(s) have uncertain significance, resistance to {} cannot be ruled out".format(drug)
-      elif annotation == "Assoc w R":
+      if annotation == "Assoc w R":
         return "Genetic determinant(s) associated with resistance to {} detected".format(drug)
-      else:
-        return "No resistance to {} detected".format(drug)
+      elif (annotation == "Assoc w R - interim") or (annotation == "Uncertain significance"):
+        return "The detected genetic determinant(s) have uncertain significance, resistance to {} cannot be ruled out".format(drug)
+      else: # "Not assoc w R" and "Not assoc w R - Interim" and anything else
+        return "No genetic determinants associated with resistance to {} detected".format(drug)
 
-    def parse_json_resistance(json_file):
+    def parse_json_resistance(json_file, destination):
+      """
+      This function parses the tbprofiler json report and returns a resistance dictionary
+      containing the WHO resistance annotation for each antimicrobial drug for LIMS and Looker. 
+      The annotation corresponds to the highest ranked one regarding severity (R > I > S)
+      """
       resistance_dict = {}
 
+
       with open(json_file) as js_fh:
         results_json = json.load(js_fh)
-        for dr_variant in results_json["dr_variants"]:  # reported mutation by tb-profiler, all confering resistance
-          for antimicrobial in dr_variant["gene_associated_drugs"]:
-            if antimicrobial not in resistance_dict.keys():
-              resistance_dict[antimicrobial] = "Assoc w R"
-            
-        for other_variant in results_json["other_variants"]:  # mutations not reported by tb-profiler
+
+        for dr_variant in results_json["dr_variants"]: # mutation 
+          name = dr_variant["gene"]
+
+          if dr_variant["type"] != "synonymous_variant":  # report all non-synonymous mutations
+
+            if "annotation" in dr_variant.keys(): # if an annotation is present,
+              if len(dr_variant["annotation"]) == 0:
+                for identified_drug in dr_variant["gene_associated_drugs"]:
+                  who_annotation = apply_expert_rules(dr_variant["nucleotide_change"], dr_variant["protein_change"], dr_variant["gene"], dr_variant["type"], "LIMS")
+
+                  if identified_drug not in resistance_dict.keys():
+                    if (name in genes_for_LIMS) and (destination == "LIMS"):
+                      resistance_dict[identified_drug] = who_annotation # overwrite with more severe annotation
+                  else:
+                    if rank_annotation(resistance_dict[identified_drug]) < rank_annotation(who_annotation): # if current annotation indicates higher severity than any previous annotation,
```

Comment:
Line is getting kind of long, consider moving the comment before or after to make easier to read